### DISCUSSION TITLE
Failed to read file `file(lamnormal)'

### DISCUSSION THREAD

**Discussion by wsshlj** [neperfepx/neper#304] [Q&A] [answered]
When using the `"lamellar(w=file(lamwidth),v=file(lamnormal))"` command, it will report an error `"Failed to read file 'file(lamnormal)'"`. I have put `lamwidth` and `lamnormal` in my operation In the directory, what is the reason for this. The neper version is 4.0.2 and the ubuntu version is 18.04LTS.

**Comment by rquey** [comment, marked as answer]
It is difficult to diagnose the problem without specifics, but this is probably caused by a wrong input...

Here is a working example:
```
$ neper -T -n 2::from_morpho -morpho "voronoi::lamellar(w=file(lamwidth.txt),v=file(lamnormal.txt))" -o foo

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.2.0
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 8 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2021, and GNU GPL'd, by Romain Quey.
Info   : Loading initialization file `/home/rquey/.neperrc'...
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] -tesrformat ascii
Info   : [com line] -n 2::from_morpho -morpho
         voronoi::lamellar(w=file(lamwidth.txt),v=file(lamnormal.txt)) -o
         foo
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellations...
Info   :   [level] Running in multiscale mode (2 scales)...
Info   :   [    1]  1/1 Info   :   - Setting seeds...                   
Info   :   - Generating crystal orientations...
Info   :   - Running tessellation...
Info   :   [    2]  1/2 Info   :   - Running tessellation...            
Info   :   - Generating crystal orientations...
Info   :   [    2]  2/2 Info   :   - Running tessellation...            
Info   :   - Generating crystal orientations...
Info   : Flattening 2-scale tessellation...
Info   :   [level] Flattening 2-scale tessellation...
Info   :   [    1]  1/1: id=1                                           
Info   : Writing results...
Info   :     [o] Writing file `foo.tess'...
Info   :     [o] Wrote file `foo.tess'.
Info   : Elapsed time: 0.011 secs.
========================================================================
```
[lamnormal.txt](https://github.com/rquey/neper/files/7418697/lamnormal.txt)
[lamwidth.txt](https://github.com/rquey/neper/files/7418698/lamwidth.txt)

**Comment by xshang93** [comment]
This happened to me too. My solution is to just create a new lamnormal file, with the same content, using a text editor and it would work fine.

**Reply by rquey** [reply]
It may be a text file format issue (DOS vs Unix). Try to run `dos2unix file_name`.

**Reply by xshang93** [reply]
by the way, please delete if this is improperiate - I wonder if you could share with me your code used for modeling the 3-scale tessellation of TI64 in your paper "The inﬂuence of mechanical constraints introduced by β annealed microstructures on the yield strength and ductility of Ti-6Al-4V"? I'm working on some research of Ti64, and it would be very helpful if you could share with me this. I have been testing it with neper for a few weeks now, and I'm getting close, but not quite yet there.. my email is 93xshang@gmail.com Thank you very much!
